src/items.c

```diff
@@ -76,90 +106,90 @@ bool8 IsThrowableItem(u8 itemIndex)
-        param_1->numItems = RandomRange(uVar3,uVar4);
+        uVar3 = GetItemUnkThrow(itemIndex, 0);
+        uVar4 = GetItemUnkThrow(itemIndex, 1);
+        slot->numItems = RandomRange(uVar3, uVar4);
     }
     else if (GetItemType(itemIndex) == ITEM_TYPE_MONEY)
-        param_1->numItems  = 1;
+        slot->numItems  = 1;
     else 
-        param_1->numItems  = 0;
+        slot->numItems  = 0;
 
     if (param_3 != 0)
-        param_1->unk0 |= 8;
+        slot->unk0 |= 8;
 
   }
   else {
-    param_1->unk0 = 0;
-    param_1->itemIndex = ITEM_ID_NOTHING;
-    param_1->numItems  = 0;
+    slot->unk0 = 0;
+    slot->itemIndex = ITEM_ID_NOTHING;
+    slot->numItems  = 0;
   }
 }
 
-void sub_8090B08(struct ItemSlot_ALT *param_1,u8 itemIndex)
+void xxx_init_helditem_8090B08(struct HeldItem *held, u8 itemIndex)
 {
   u32 uVar2;
   u32 uVar3;
   
   if (itemIndex != ITEM_ID_NOTHING) {
-    param_1->itemIndex = itemIndex;
+    held->itemIndex = itemIndex;
     if (IsThrowableItem(itemIndex)) {
         uVar2 = GetItemUnkThrow(itemIndex,0);
         uVar3 = GetItemUnkThrow(itemIndex,1);
-        param_1->numItems = RandomRange(uVar2,uVar3);
+        held->numItems = RandomRange(uVar2, uVar3);
     } 
     else if (GetItemType(itemIndex) == ITEM_TYPE_MONEY)
-        param_1->numItems = 1;
+        held->numItems = 1;
     else
-        param_1->numItems = 0;
+        held->numItems = 0;
   }
   else {
-    param_1->itemIndex = 0;
-    param_1->numItems = 0;
+    held->itemIndex = 0;
+    held->numItems = 0;
   }
 }
 
-void sub_8090B64(struct ItemSlot *param_1, struct ItemSlot_ALT *param_2)
+void HeldItemToSlot(struct ItemSlot *slot, struct HeldItem *held)
 {
-    u8 r6;
+    u8 is_throwable;
 
-    if(param_2->itemIndex != ITEM_ID_NOTHING)
+    if(held->itemIndex != ITEM_ID_NOTHING)
     {
-        param_1->unk0 = 1;
-        param_1->itemIndex = param_2->itemIndex;
-        r6 = IsThrowableItem(param_1->itemIndex);
-        if(r6 != 0 || GetItemType(param_1->itemIndex) == ITEM_TYPE_MONEY)
-            param_1->numItems = param_2->numItems;
-        else if(param_1->itemIndex == ITEM_ID_USED_TM)
-            param_1->numItems = param_2->numItems;
+        slot->unk0 = 1;
+        slot->itemIndex = held->itemIndex;
+        is_throwable = IsThrowableItem(slot->itemIndex);
+        if(is_throwable != 0 || GetItemType(slot->itemIndex) == ITEM_TYPE_MONEY)
+            slot->numItems = held->numItems;
+        else if(slot->itemIndex == ITEM_ID_USED_TM)
+            slot->numItems = held->numItems;
         else
-            param_1->numItems = 0;
+            slot->numItems = 0;
     }
     else
     {
-        param_1->itemIndex = ITEM_ID_NOTHING;
-        param_1->numItems = 0;
-        param_1->unk0 = 0;
+        slot->itemIndex = ITEM_ID_NOTHING;
+        slot->numItems = 0;
+        slot->unk0 = 0;
     }
 }
 
-void sub_8090BB0(struct ItemSlot_ALT *param_1,struct ItemSlot *param_2)
+void SlotToHeltItem(struct HeldItem *held,struct ItemSlot *slot)
```

Comment:
```suggestion
void SlotToHeldItem(struct HeldItem *held,struct ItemSlot *slot)
```
Or is that name intentional?